cmd/clusterctl/client/cluster/components_test.go

```diff
@@ -42,7 +42,11 @@ func Test_providerComponents_Delete(t *testing.T) {
 	crd.SetKind("CustomResourceDefinition")
 	crd.SetName("crd1")
 	crd.SetLabels(labels)
-
+	providerInventory := unstructured.Unstructured{}
```

Comment:
Q: is there a particular reason why you are using unstructured instead of using the typed object? (clusterctlv1 is already imported in this package)

Reply:
This is part of the issue I was mentioning above. The clusterctlv1.Provider kind doesn't have the Group or Kind by default so they have to be set. This test uses the Provider type to represent the actual providers being deleted (without using the provider type) so there's ambiguity as to which is which. I chose to go unstructured here as the lowest lift way of dealing with that issue.